### DISCUSSION TITLE
By the way: Simple Ansible reporting with ara

### DISCUSSION THREAD

**Discussion by dmsimard** [O-X-L/ansible-webui-v0#7] [Ideas]
Hi, nice project!

I like simplicity and think it is a feature.
I just wanted to share a simple way to get reporting with Ansible: https://github.com/ansible-community/ara
Live demo here: https://demo.recordsansible.org/

Maybe it can be useful or give you ideas, it has an API but isn't in the business of running playbooks, it only records them.

Feel free to reach out if you have questions.

**Comment by superstes** [comment]
Yeah - could make sense to add an 'ARA reporting server' option to the system config and if set - add it to all Ansible executions

```
# Configure Ansible to enable ara
export ANSIBLE_CALLBACK_PLUGINS="$(python3 -m ara.setup.callback_plugins)"

# Set up the ara callback to know where the API server is located
export ARA_API_CLIENT="<FROM-CONFIG>"
export ARA_API_SERVER="<FROM-CONFIG>"
```

The ara module would have to be installed as seen in the docs.
```
python3 -m pip install ara
```
This would be useful for people that want to have deeper insights into their execution stats.

**Comment by ansibleguy** [comment]
I'll test it - but users should already be able to set a 'global' ARA server for all executions by exporting the ENV-VARS before starting ansibleguy-webui.
An option in the system-settings could make sense. Maybe a 'global job env-vars' setting would be beneficial so users could enable such features on their own as needed.

Including an ARA server in this product is out-of-scope in my mind.

**Reply by ansibleguy** [reply]
Will be available in the next release: https://github.com/ansibleguy/webui/commit/415edc13276cd45659496524e4f75bda1126abff
